### DISCUSSION TITLE
Can we represent loops in the rules?

### DISCUSSION THREAD

**Discussion by karthik6244** [microsoft/RulesEngine#121] [Q&A]
Say a complex object is there I want to iterate over that object and apply rule on one property, is this possible?

"Rules": [
      {
        "RuleName": "GiveDiscount10",
        "SuccessEvent": "10",
        "ErrorMessage": "One or more adjust rules failed.",
        "ErrorType": "Error",
        "RuleExpressionType": "LambdaExpression",
        "Expression": "**_Loop through Complexobject AND input1.country == \"india\"_** AND input1.loyalityFactor <= 2 AND input1.totalPurchasesToDate >= 5000"
      },

**Comment by abbasc52** [comment]
Assuming you are using at list or array.
You can use linq method such as `Any, All , First, etc` to check for some condition for items of array.

As for full-fledged support for looping, it is not possible via expression. You can write a utils class with a method to do looping for you and inject it in RulesEngine. You can refer it here - https://github.com/microsoft/RulesEngine/wiki/Getting-Started#resettings
